sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -757,7 +758,10 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = {
+    val path = new Path(getConf(WAREHOUSE_PATH))
+    FileSystem.get(path.toUri, new Configuration()).makeQualified(path).toString
```

Comment:
@avulanov just so I'm testing the right things, what would a user set for the warehouse dir, on Windows, if they wanted to specify a local path? It would have to be like `file:///C:/...` right? That seems to be correctly resolved. That is, we can't expect "C:/..." or "C:\..." to work, because we need this to be parsed as a URI or relative path -- is that right? If so then I suspect @vanzin is right, let me try that approach.